airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceConfigurationJsonObject.kt

```diff
@@ -232,29 +232,68 @@ class SslVerifyCertificate : Encryption {
     @JsonSchemaInject(json = """{"airbyte_secret":true,"multiline":true}""")
     lateinit var sslCertificate: String
 
-    @JsonProperty("ssl_client_certificate")
+    @JsonProperty("ssl_client_certificate", required = false)
     @JsonSchemaTitle("Client certificate File")
     @JsonPropertyDescription(
         "Client certificate (this is not a required field, but if you want to use it, you will need to add the Client key as well)",
     )
     @JsonSchemaInject(json = """{"airbyte_secret":true,"multiline":true}""")
-    lateinit var sslClientCertificate: String
+    var sslClientCertificate: String? = null
 
     @JsonProperty("ssl_client_key")
     @JsonSchemaTitle("Client Key")
     @JsonPropertyDescription(
         "Client key (this is not a required field, but if you want to use it, you will need to add the Client certificate as well)",
     )
     @JsonSchemaInject(json = """{"airbyte_secret":true,"multiline":true}""")
-    lateinit var sslClientKey: String
+    var sslClientKey: String? = null
 
     @JsonProperty("ssl_client_key_password")
     @JsonSchemaTitle("Client key password")
     @JsonPropertyDescription(
         "Password for keystorage. This field is optional. If you do not add it - the password will be generated automatically.",
     )
     @JsonSchemaInject(json = """{"airbyte_secret":true,"multiline":true}""")
-    lateinit var sslClientPassword: String
+    var sslClientPassword: String? = null
+}
+
+@JsonSchemaTitle("verify_identity")
+@JsonSchemaDescription(
+    "To always require encryption and verify that the source has a valid SSL certificate."
+)
+@SuppressFBWarnings(value = ["NP_NONNULL_RETURN_VIOLATION"], justification = "Micronaut DI")
+class SslVerifyIdentity : Encryption {
```

Comment:
@postamar I'm actually not sure what's the best practice there. SslVerifyIdentity shares the same members as SslVerifyCertificate, but they just match to different JsonSchemaTitle and description. 

Here I simply copied them. Should I have class SslVerifyIdentity extending from SslVerifyCertificate instead?

Reply:
Copying is OK here. This code is not going to change much, ever. If it's really unreadable, then using `sealed class` to abstract shared components is OK. I don't know how well that works with the annotations, though.

Reply:
The question is, should these objects also be in the CDK, like SSH tunneling? Are they common patterns? Let's look into this once we've migrated all the existing certified sources.

Reply:
They may share some fields but most connectors have pretty different settings regarding to SSL

Reply:
Thanks! Good to know :+1: